.gitbook/3-protocol-devs/1-dep-specs/5-confidential-oracle.md

```diff
@@ -0,0 +1,50 @@
+# Confidential Oracle
+
+- Status: Draft
+- Created: 2023-01-06
+- Modified: 2023-01-06
+- Authors
+  - Inchul Song <icsong@medibloc.org>
+  - Youngjoon Lee <yjlee@medibloc.org>
+  - Gyuguen Jang <gyuguen.jang@medibloc.org>
+  - Hansol Lee <hansol@medibloc.org>
+  - Myongsik Gong <myongsik_gong@medibloc.org>
+
+
+## Synopsis
+This document explains that why the oracles must use confidential computing.
+
+## Privacy Vulnerabilities of Oracle
+In DEP, anyone can operate oracle node. However, when validating the sensitive data through the general-purpose oracle, it could be occurred several vulnerabilities.
+The one of vulnerabilities is that the sensitive data could be exposed to oracle operator who uses malicious binary.
+The other vulnerabilities is that the data could be stolen from external node or attack.
+So preventing from these vulnerabilities, the oracle must use confidential computing by using Intel SGX Technology in [TEE(Trusted Execution Environment)](https://en.wikipedia.org/wiki/Trusted_execution_environment).
+
+## Confidential Oracle
+In DEP, anyone can operate oracle nodes but must use confidential computing.
+If the operator runs the confidential oracle, the only oracle software can decrypt a data and validate it. It means that the oracle operator can't decrypt a data and it will be not exposed to the operator.
+Also, when running the confidential oracle, the only genuine binary built in Intel SGX should be used so that it can be prevented from the node who uses malicious binary. It means that all oracles run the same binary.
+Using a genuine binary can preserve a privacy of data.
+
+### Additional Effort
+
+Using a confidential oracle could reinforce a security, but we developed an additional effort for protecting a data leak.
+When running an oracle with the genuine binary, the data encryption and decryption are performed by oracle private key which is sealed by unique ID located in Intel SGX.
+So if the binary is changed or corrupted, the data encryption and decryption would not work.
```

Comment:
I'm not sure the word `corrupted` is right.

Reply:
`tampered` would be good.